logging/zerolog/option.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 CloudWeGo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zerolog
+
+import (
+	"errors"
+
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type ExtraKey string
+
+type Option interface {
+	apply(cfg *config)
+}
+
+type option func(cfg *config)
+
+func (fn option) apply(cfg *config) {
+	fn(cfg)
+}
+
+type traceConfig struct {
+	recordStackTraceInSpan bool
+	errorSpanLevel         zerolog.Level
+}
+
+type config struct {
+	logger      *zerolog.Logger
+	traceConfig *traceConfig
+}
+
+// defaultConfig default config
+func defaultConfig() *config {
+	return &config{
+		traceConfig: &traceConfig{
+			recordStackTraceInSpan: true,
+			errorSpanLevel:         zerolog.ErrorLevel,
+		},
+	}
+}
+
+// WithLogger configures logger
+func WithLogger(logger *zerolog.Logger) Option {
+	return option(func(cfg *config) {
+		cfg.logger = logger
+	})
+}
+
+// WithTraceErrorSpanLevel trace error span level option
+func WithTraceErrorSpanLevel(level zerolog.Level) Option {
+	return option(func(cfg *config) {
+		cfg.traceConfig.errorSpanLevel = level
+	})
+}
+
+// WithRecordStackTraceInSpan record stack track option
+func WithRecordStackTraceInSpan(recordStackTraceInSpan bool) Option {
+	return option(func(cfg *config) {
+		cfg.traceConfig.recordStackTraceInSpan = recordStackTraceInSpan
+	})
+}
+
```

Comment:
In theory, this library is a combination of the "logger" and "obs logger" libraries, so it should also provide a configuration related to the logger.
https://github.com/hertz-contrib/logger/blob/main/zerolog/options.go

Reply:
I can't find the similar `logger` in `kitex-contrib`.

I also checked on other logger libraries in this repository; and I see that we also want to skip the logger like `hertz`. I don't know if we have any historical decisions related to this, but I try to follow it.

Reply:
Thanks.
We have split the log library of Hertz, but it has not been split here temporarily.
So this library actually includes the functionalities of two libraries from Hertz.